deployment/deployment.go

```diff
@@ -0,0 +1,1224 @@
+			currentimage = ""
+		} else {
+			utils.ReportError(err, r)
+			return
+		}
+	}
+
+	deployment.Image = structs.PrettyNullString{NullString: sql.NullString{
+		String: currentimage,
+		Valid:  currentimage != "",
+	}}
+
+	r.JSON(http.StatusOK, deployment)
+}
+
+func getConfigVars(db *sql.DB, appname string, space string) ([]structs.EnvVar, error) {
+	// Get bindings
+	appconfigset, appbindings, err := config.GetBindings(db, space, appname)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get user defined config vars
+	configvars, err := config.GetConfigVars(db, appconfigset)
+	if err != nil {
+		return nil, err
+	}
+
+	// Assemble config -- akkeris "built in config", "user defined config vars", "service configvars"
+	elist := app.AddAkkerisConfigVars(appname, space)
+	for n, v := range configvars {
+		elist = append(elist, structs.EnvVar{Name: n, Value: v})
+	}
+
+	// add service vars
+	err, servicevars := service.GetServiceConfigVars(db, appname, space, appbindings)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, e := range servicevars {
+		elist = append(elist, e)
+	}
+	return elist, nil
+}
+
+// GetAllConfigVarsV2 - Get all config vars for a deployment
+func GetAllConfigVarsV2(db *sql.DB, params martini.Params, r render.Render) {
+	name := params["deployment"]
+	space := params["space"]
+
+	exists, err := checkDeployment(db, name, space)
+	if err != nil {
+		utils.ReportError(err, r)
+		return
+	}
+	if exists == false {
+		utils.ReportNotFoundError(r)
+		return
+	}
+
+	configList, err := getConfigVars(db, name, space)
+	if err != nil {
+		utils.ReportError(err, r)
+		return
+	}
+
+	r.JSON(201, configList)
+}
+
+// AddDeploymentV2 - V2 version of space.AddApp
+// (original: "space/app.go")
+func AddDeploymentV2(db *sql.DB, params martini.Params, deployment structs.AppDeploymentSpec, berr binding.Errors, r render.Render) {
+	if berr != nil {
+		utils.ReportInvalidRequest(berr[0].Message, r)
+		return
+	}
+
+	if !deployment.AppID.Valid || deployment.AppID.String == "" {
+		utils.ReportInvalidRequest("App ID is required", r)
+		return
+	}
+
+	name := params["deployment"]
+	space := params["space"]
+
+	exists, err := checkDeployment(db, name, space)
+	if err != nil {
+		utils.ReportError(err, r)
+		return
+	}
+
+	if exists {
+		utils.ReportInvalidRequest("Deployment already exists", r)
+		return
+	}
+
+	var healthcheck *string
+	if deployment.Healthcheck.String == "" {
+		healthcheck = nil
+	} else {
+		healthcheck = &deployment.Healthcheck.String
+	}
+
+	insertQuery := "insert into v2.deployments(name, space, plan, instances, healthcheck, appid) values($1, $2, $3, $4, $5, $6) returning name"
+	inserterr := db.QueryRow(insertQuery, name, space, deployment.Plan, int(deployment.Instances.Int64), healthcheck, deployment.AppID.String).Scan(&name)
+	if inserterr != nil {
+		utils.ReportError(inserterr, r)
+		return
+	}
+
+	r.JSON(http.StatusCreated, structs.Messagespec{Status: http.StatusCreated, Message: "Deployment record created"})
```

Comment:
I'm trying to deprecate the use of a `structs.Messagespec` as it has very little value.  Can we consider returning back the deployment object?